analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ReturnEmptyCollectionInsteadOfNull.CSharp9.cs

```diff
@@ -28,3 +29,82 @@ IEnumerable<char> Method(string str)
 
     IEnumerable<int> Compliant() => Enumerable.Empty<int>();
 }
+
+class TernariesAndDefaults
+{
+    private static readonly string checkAgainst = "42";
+
+    public static IEnumerable<string> Foo1() => null; // Noncompliant
+    public static IEnumerable<string> Foo2() => (null); // Noncompliant
+    public static IEnumerable<string> Foo3() => default; // Noncompliant
+    public static IEnumerable<string> Foo4() => (default); // Noncompliant
+    public static IEnumerable<string> Foo5() => true ? null : new List<string>(); // Noncompliant
+    public static IEnumerable<string> Foo6() =>
+        ((
+        checkAgainst == null // Compliant
+        ? default // Noncompliant
+        : checkAgainst is not ((null)) // Compliant
+            ? ((default)) // Secondary
+            : DateTime.Now.Second % 2 == 0
+                ? null // Secondary
+                : (((null))) // Secondary
+        ));
+
+    public static IEnumerable<string> Foo7()
+    {
+        if (true)
+        {
+            if (true)
+            {
+                if (true)
+                {
+                    return ((null)); // Noncompliant
+                }
+            }
+            else
+            {
+                return default(IEnumerable<string>); // Secondary
+            }
+        }
+        else
+        {
+            return checkAgainst != null // Compliant
+                ? new List<string>()
+                : (default); // Secondary
+        }
+    }
+
+    public static IEnumerable<string> Foo8()
+    {
+        return true
+            ? new List<string>()
+            : false
+                ? default // Noncompliant
+                : null;  // Secondary 
+    }
+}
+
+// Reproducer for #6491 https://github.com/SonarSource/sonar-dotnet/issues/6491
+namespace Issue_6491
```

Comment:
These examples are not C#9 related. Please move them to the proper versioned test cases.